Managers/FrameMan.cpp

```diff
@@ -124,14 +128,75 @@ namespace RTE {
 		} else if ((resX * resMultiplier) % 4 > 0) {
 			allegro_message("Resolution width that is not divisible by 4 is not supported!\nOverriding to closest valid width!");
 			resX = m_NewResX = std::floor(resX / 4) * 4;
-		} else {
+		}
+
+		if (m_NumScreens == 1) {
 			float currentAspectRatio = static_cast<float>(resX) / static_cast<float>(resY);
 			if (currentAspectRatio < 1 || currentAspectRatio > 4) {
 				allegro_message("Abnormal aspect ratio detected! Reverting to defaults!");
 				resX = m_NewResX = 960;
 				resY = m_NewResY = 540;
 				resMultiplier = m_ResMultiplier = m_NewResMultiplier = 1;
 			}
+		} else if (!m_DisableMultiScreenResolutionValidation && m_NumScreens > 1 && m_NumScreens < 4) {
+			if (resX * resMultiplier > m_PrimaryScreenResX || resY * resMultiplier > m_PrimaryScreenResY) { ValidateMultiScreenResolution(resX, resY, resMultiplier); }
+		} else if (!m_DisableMultiScreenResolutionValidation && m_NumScreens > 3) {
+			allegro_message("Number of screens is too damn high! Overriding to defaults!\n\nPlease disable multi-screen resolution validation in \"Settings.ini\" and run at your own risk!");
+			resX = m_NewResX = 960;
+			resY = m_NewResY = 540;
+			resMultiplier = m_ResMultiplier = m_NewResMultiplier = 1;
+		}
+
+		g_SettingsMan.UpdateSettingsFile();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void FrameMan::ValidateMultiScreenResolution(unsigned short &resX, unsigned short &resY, unsigned short &resMultiplier) {
+		POINT pointOnScreen;
+		HMONITOR screenHandle;
+		MONITORINFO screenInfo;
+
+		pointOnScreen = { -1 , 0 };
+		screenHandle = MonitorFromPoint(pointOnScreen, MONITOR_DEFAULTTONULL);
+		if (screenHandle != NULL) {
+			const char *leftNotPrimaryMessage = {
+				"Due to limitations in Cortex Command's graphics API it is impossible to properly run multi-screen mode when the left-most screen is not set as primary.\n"
+				"Please configure your left-most screen to be primary to utilize all screens, as the game window will extend right but will not extend left, leaving any screen left of the primary unused.\n\n"
+				"You can disable multi-screen resolution validation in \"Settings.ini\" and run at your own risk!\n\nResolution settings will be overridden fit primary screen only!"
```

Comment:
Typo - "Resolution settings will be overridden fit primary..." should be "Resolution settings will be overridden to fit primary..."

Also, I'm a little confused how this char* works, You're using list initializer {}s, but there's no commas between lines and the strings aren't concatenated with +s?? Anyway, I think in general (and in this case) we should lean towards using cpp std::strings and getting there c_str() when we need the c-style string. It's a little less direct, so where it's an issue of constant performance, maybe not (assuming std::strings aren't somehow faster to make or something), but c-style strings are awful and we should minimize our use of them as much as we can.